doc/architectural/memory-primitives.md

```diff
@@ -91,6 +91,41 @@ point into another memory object (such as could happen when running on a real
 machine). To verify that pointers stay within the bounds of their pointees, the
 CBMC option `--pointer-overflow-check` can be used.
 
+### Malloc modelling
+
+CBMC ships a model of `malloc` that seeks to emulate the behaviour of the C
+standard library. By default, however, it will always succeed to allocate an
+object. This extends to the case of `malloc(0)`, which returns a valid pointer
```

Comment:
Do you want to give a bit of rationale, and also that this might miss the case where `malloc` returns `NULL`?

Reply:
I have reworded this.